jsonpickle/util.py

```diff
@@ -43,10 +43,19 @@ def is_type(obj):
     >>> is_type(Klass)
     True
     """
+    # use "isinstance" and not "is" to allow for metaclasses
     if PY3:
-        return type(obj) is type
+        return isinstance(obj, type)
     else:
-        return type(obj) is type or type(obj) is types.ClassType
+        return isinstance(obj, (type, types.ClassType))
+
+
+def has_method(obj, name, *args, **kwargs):
+    try:
+        getattr(obj, name)(*args, **kwargs)
+        return True
+    except:
+        return False
```

Comment:
It looks like this `has_method()` predicate also has the side-effect of actually calling the method. That can have strange side-effects in user code and it's a bit wasteful.

How about reworking this to check whether the attribute is callable. You'll need to adjust the tests, but maybe something like:

    def has_callable(obj, name):
        return callable(getattr(obj, name))

...and then use `has_callable()` instead of `has_method()`.  When the attribute doesn't exist we'll get `None` from `getattr()` and `callable()` will return `False`.